src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -119,9 +119,15 @@ public void run(ActionContext context) throws Exception {
     String tableName = config.getTable();
     String datasetProjectId = config.getDatasetProject();
     if (!Strings.isNullOrEmpty(datasetName) && !Strings.isNullOrEmpty(tableName)) {
-      BigQuerySinkUtils.createDataset(bigQuery, DatasetId.of(datasetProjectId, datasetName), config.getLocation(),
-                                      cmekKeyName, () -> String.format("Unable to create BigQuery dataset '%s.%s'",
-                                                                       datasetProjectId, datasetName));
+      DatasetId datasetId = DatasetId.of(datasetProjectId, datasetName);
```

Comment:
would it be better to put those logic to `BigQuerySinkUtils.createDataset` to make the operation idempotent ?

Reply:
We always call this method with the assumption that this check has been made

Reply:
does that mean we need to check everywhere before calling this method ?

Reply:
would it be better to do check in one single place ?

Reply:
also checking in single place will avoid future bugs like this one.

Reply:
+1 to Sean, a second thought is it the correct behaviour that the util class method depends on the caller?

Reply:
Discussed offline with @fernst to make that method back to private and add another util method `createDatasetIfNotExists`.